Source/WebCore/platform/graphics/ca/TileController.cpp

```diff
@@ -629,18 +730,21 @@ void TileController::tileSizeChangeTimerFired()
 
 IntSize TileController::tileSize() const
 {
-    return tileGrid().tileSize();
+    const TileGrid& grid = m_asyncTileConfigurationChangeData ? *m_asyncTileConfigurationChangeData->pendingTileGrid : tileGrid();
+    return grid.tileSize();
 }
 
 FloatRect TileController::rectForTile(TileIndex tileIndex) const
 {
-    return tileGrid().rectForTile(tileIndex);
+    const TileGrid& grid = m_asyncTileConfigurationChangeData ? *m_asyncTileConfigurationChangeData->pendingTileGrid : tileGrid();
```

Comment:
Hmm, we're in an ambiguous situation now; maybe the callers need the rect in the existing grid (if we allow it to remain "live"). Another option is to explicitly specify that the old grid becomes a zombie and won't continue repainting.